File: android/framework_old/src/main/java/com/dreampany/network/data/model/Network.java
/*
package com.dreampany.network.data.model;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;
import android.os.Parcel;
import android.os.Parcelable;
import androidx.annotation.NonNull;

import com.dreampany.network.data.enums.NetworkType;
import com.dreampany.network.misc.Constants;

*/
/**
 * Created by Hawladar Roman on 8/18/2018.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 *//*

@Entity(indices = {@Index(value = {"bssid", "ssid"}, unique = true)},
        primaryKeys = {"bssid", "ssid"})
public class Network implements Parcelable {

    private NetworkType type;
    private String bssid;
    private String ssid;
    private String capabilities;
    private boolean enabled;
    private boolean connected;
    private boolean internet;

    @Ignore
    public Network() {

    }

    public Network(NetworkType type) {
        this.type = type;
    }

    @Ignore
    private Network(Parcel in) {

    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {

    }

    public static final Creator<Network> CREATOR = new Creator<Network>() {
        @Override
        public Network createFromParcel(Parcel in) {
            return new Network(in);
        }

        @Override
        public Network[] newArray(int size) {
            return new Network[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder("network#");
        builder.append(" type: ").append(type);
        builder.append(" bssid: ").append(bssid);
        builder.append(" ssid: ").append(ssid);
        builder.append(" open: ").append(isOpen());
        builder.append(" enabled: ").append(enabled);
        builder.append(" connected: ").append(connected);
        builder.append(" internet: ").append(internet);
        return builder.toString();
    }

    public void setType(NetworkType type) {
        this.type = type;
    }

    public void setBssid(String bssid) {
        this.bssid = bssid;
    }

    public void setSsid(String ssid) {
        this.ssid = ssid;
    }

    public void setCapabilities(String capabilities) {
        this.capabilities = capabilities;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public void setConnected(boolean connected) {
        this.connected = connected;
    }

    public void setInternet(boolean internet) {
        this.internet = internet;
    }

    public NetworkType getType() {
        return type;
    }

    public String getBssid() {
        return bssid;
    }

    public String getSsid() {
        return ssid;
    }

    public String getCapabilities() {
        return capabilities;
    }

    public boolean isOpen() {
        return capabilities == null ||
                !(capabilities.contains(Constants.SECURITY.WEP) ||
                        capabilities.contains(Constants.SECURITY.PSK) ||
                        capabilities.contains(Constants.SECURITY.EAP));
    }

    public boolean isEnabled() {
        return enabled;
    }

    public boolean isConnected() {
        return connected;
    }

    public boolean hasInternet() {
        return internet;
    }
}
*/

File: android/framework_old/src/main/kotlin/com/dreampany/network/data/model/Network.kt
package com.dreampany.network.data.model

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.Index
import com.dreampany.network.misc.Constants
import com.google.common.base.Objects
import com.google.firebase.database.IgnoreExtraProperties
import kotlinx.android.parcel.Parcelize

/**
 * Created by Roman-372 on 6/27/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */

@Parcelize
@IgnoreExtraProperties
@Entity(
    indices = [Index(value = [Constants.Network.BSSID, Constants.Network.SSID], unique = true)],
    primaryKeys = [Constants.Network.BSSID, Constants.Network.SSID]
)
data class Network(val type: Type) : Parcelable {

    enum class Type {
        WIFI, HOTSPOT, BLUETOOTH;
    }

    var bssid: String //only for hotspot
    var ssid: String //only for hotspot
    var capabilities: String? = null
    var enabled: Boolean = false
    var connected: Boolean = false
    var internet: Boolean = false

    init {
        bssid = ""
        ssid = ""
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as Network
        return Objects.equal(bssid, item.bssid) && Objects.equal(ssid, item.ssid)
    }

    override fun hashCode(): Int {
        return Objects.hashCode(bssid, ssid)
    }

    override fun toString(): String {
        val builder = StringBuilder("network#");
        builder.append(" type: ").append(type)
        builder.append(" bssid: ").append(bssid)
        builder.append(" ssid: ").append(ssid)
        builder.append(" open: ").append(isOpen())
        builder.append(" enabled: ").append(enabled)
        builder.append(" connected: ").append(connected)
        builder.append(" internet: ").append(internet)
        return builder.toString();
    }

    fun isOpen(): Boolean {
        capabilities?.let {
            return !(it.contains(Constants.SECURITY.WEP) ||
                    it.contains(Constants.SECURITY.PSK) ||
                    it.contains(Constants.SECURITY.EAP))
        }
        return true
    }
}